### PR TITLE
[eas-cli] Update build metadata for internal builds

### PR DESCRIPTION
**Pull request by wkozyra95**
# Checklist

- [ ] I've added an entry to [CHANGELOG.md](https://github.com/expo/eas-cli/blob/main/CHANGELOG.md#main) if necessary. You can comment this pull request with `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]` and CHANGELOG.md will be updated automatically.

# Why

Update metadata from internal builds

# How

Use new graphql endpoint to update metadata from `eas build:internal` command

# Test Plan

Tested with local www

### CODE REVIEW

File: packages/eas-cli/src/build/build.ts [hunk 1]
```diff
@@ -142,6 +143,10 @@ export async function prepareBuildRequestForPlatformAsync<
       await runLocalBuildAsync(job, metadata, ctx.localBuildOptions);
       return undefined;
     } else if (ctx.localBuildOptions.localBuildMode === LocalBuildMode.INTERNAL) {
+      await BuildMutation.updateBuilMetadatadAsync(ctx.graphqlClient, {
+        buildId: nullthrows(process.env.EAS_BUILD_ID),
```
Comment (wkozyra95) [inline packages/eas-cli/src/build/build.ts:147]: This env is defined in EAS Build env

File: packages/eas-cli/src/graphql/mutations/BuildMutation.ts [hunk 2]
```diff
@@ -119,6 +121,34 @@ export const BuildMutation = {
     );
     return nullthrows(data.build?.createIosBuild);
   },
+  async updateBuilMetadatadAsync(
```
Comment (dsokal) [inline packages/eas-cli/src/graphql/mutations/BuildMutation.ts:124]:
```suggestion
  async updateBuildMetadataAsync(
```

File: packages/eas-cli/src/graphql/mutations/BuildMutation.ts [hunk 3]
```diff
@@ -119,6 +121,34 @@ export const BuildMutation = {
...
+                updateBuildMetadata(buildId: $buildId, metadata: $metadata) {
+                  id
+                  ...BuildFragment
+                }
+              }
+            }
+            ${print(BuildFragmentNode)}
+          `,
+          input,
+          { noRetry: true }
```
Comment (dsokal) [inline packages/eas-cli/src/graphql/mutations/BuildMutation.ts:146]: Is this necessary?  I don't have an opinion, just asking.

**Review by dsokal — COMMENTED** [review]
LGTM

File: packages/eas-cli/src/graphql/mutations/BuildMutation.ts (hunk 3 above)
Comment (wkozyra95) [reply packages/eas-cli/src/graphql/mutations/BuildMutation.ts:146]: no, the update should be idempotent

**Review by szdziedzic — APPROVED** [review]

**Review by dsokal — APPROVED** [review]

File: packages/eas-cli/src/update/__tests__/getBranchNameFromChannelNameAsync-test.ts [hunk 4]
```diff
@@ -121,6 +121,7 @@ function mockUpdateBranches(branchNames: string[]): UpdateBranch[] {
           message: 'my message',
           createdAt: '2022-12-07T02:24:43.487Z',
           runtimeVersion: '1.0.0',
+          runtime: {} as Runtime, // Temporary fix to resolve type errors
```
Comment (wkozyra95) [inline packages/eas-cli/src/update/__tests__/getBranchNameFromChannelNameAsync-test.ts:124]: This change was introduced in graphql API in https://github.com/expo/universe/commit/3e1eec1811f3256c5e8d5dcb5ba6a3cb6ea80f7c
Not sure how it should be addressed, so for now just casting it 

cc @quinlanj

**Comment by wkozyra95** [comment]
/changelog-entry chore Update metadata on the worker when using git-based builds.

**Pull request merged**